rasterio/_base.pyx

```diff
@@ -1449,6 +1451,27 @@ cdef class DatasetBase:
         rpcs = value.to_gdal()
         self._set_rpcs(rpcs)
 
+    def _xygrid(self):
+        """Compute a pair of arrays from an affine transform matrix.
+
+        TODO: results are undefined if the dataset has only GCPs or
+        RPCs.
+
+        Values correspond to the upper left corner of raster cells.
+
+        Returns
+        -------
+        tuple
+            A pair of geolocation coordinate arrays. The first is "x"
+            (easting) coordinates of raster cell centers, the second is
+            "y" (northing) coordinates.
+        """
+        cols, rows = numpy.meshgrid(range(self.width), range(self.height))
```

Comment:
Do you think this would be more efficient?
```python
        transform = self.transform * self.transform.translation(0.5, 0.5)  # adjust to cell center
        xs, ys = transform * np.meshgrid(
            np.arange(self.width),
            np.arange(self.height),
        )
```

Reply:
Yes, that's better, and so simple that it makes me think we don't even need a method for this.

Reply:
I think that it is worthwhile - if only to help users find the optimal way to do this operation.

Reply:
I deleted it for now, may bring it back once we've done a bit more discovery.